src/components/Login/utils.ts

```diff
@@ -22,6 +22,9 @@ export const decodeQr = (code: string): DecodedQrResponse => {
   } catch (e) {
     if (e.message.includes("Unexpected token"))
       throw new Error(ERROR_MESSAGE.AUTH_FAILURE_EXPIRED_TOKEN);
+    // if user is not an operator
+    if (e.message.includes("Unauthorized auth token"))
+      throw new Error(ERROR_MESSAGE.AUTH_FAILURE_INVALID_TOKEN);
```

Comment:
hmm in the decode QR stage, it is not possible to tell the role of the user, so i don't think this block should be here

Reply:
got it, removed!